packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -167,6 +167,31 @@ class FileAccessLog extends AccessLog {
   }
 }
 
+/**
+ * Represents the properties needed to define client policy
+ */
+export interface TlsClientPolicy {
+  /**
+   * Whether the policy is enforced.
+   *
+   * @default: true
+   */
+  readonly enforce?: boolean;
+
+  /**
+   * TLS is enforced on the ports specified here.
+   * If no ports are specified, TLS will be enforced on all the ports.
+   *
+   * @default: none
+   */
+  readonly ports?: number[];
+
+  /**
+   * Represents the object for TLS validation context
+   */
+  readonly tlsValidationContext: TlsValidationContext;
```

Comment:
I don't like this property. This is needless nesting that doesn't add anything to the API. Remove this, and instead add a `readonly trust: TlsValidationContextTrust;` property.

Reply:
This was decided to isolate the works in steps. To be little more in detail:
**Step1:** Create TLS validation object since it is going to be used by TLS Listener in addition to TLS client policy (*I am here)
**Step2:** Update TLS certificate object since it contains `mode` property that is actually the property for TLS Listener. In addition, TLS certificate is going to be used by TLS client policy.
**Step3:** Implement mTLS feature. `subjectiveAlternativeNames` addition is specific to mTLS feature.

Reply:
Ah. So what you're saying is that `TlsValidationContext` will be used in 2 separate places - in `TlsClientPolicy`, and then somewhere else?

Reply:
Yes, that is exactly correct. Please refer [AWS::AppMesh::VirtualNode ListenerTls](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualnode-listenertls.html) and [AWS::AppMesh::VirtualNode ClientPolicyTls](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualnode-clientpolicytls.html) as an example. As you may see, both objects have properties `certificate` and `validation`.

Reply:
OK. You've convinced me 🙂. Let's keep this interface, but rename it to `TlsValidation` (I don't think the word "context" adds much value here).